finetune/target_models/masked_language_model.py

```diff
@@ -96,6 +98,34 @@ def __init__(self, *args, **kwargs):
     def _get_input_pipeline(self):
         return MaskedLanguageModelPipeline(self.config)
 
+    def predict_top_k_report(self, input_text, k=5, context=None, **kwargs):
+        prediction_info = self._inference(
+                [input_text],
+                predict_keys=[
+                    PredictMode.GENERATE_TEXT,
+                    PredictMode.MLM_IDS,
+                    PredictMode.MLM_POSITIONS],
+                context=context,
+                force_build_lm=True,
+                **kwargs)
+
+        predicted_tokens = [{'prediction_ids': [self.input_pipeline.text_encoder.decode([i]) for i in pred['GEN_TEXT'][-k:][::-1]],
```

Comment:
Probably more direct to have just argsorted in descending order so you dont need to reverse the predictions here?

Reply:
It's a matter of where we prefer the code to be. This is a bit lazier, and there isn't an easier way of doing descending sort with np.argsort (so this code would just be where we set the GEN_TEXT key).